### PR TITLE
[W5.11][W13-A1] Ang Jing Zhe

### PR DESCRIPTION
**Pull request by AJZ1995**
2 features: Adding sort command for addressbook & Add case-insensitivity to findCommand. 

Updated minor compilation errors for previous LOs.

### CODE REVIEW

File: src/seedu/addressbook/commands/AddCommand.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import java.util.HashSet;
 import java.util.Set;
+import java.lang.*;
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/AddCommand.java:5]: Don't import everything using *; import only the necessary classes explicitly.

File: src/seedu/addressbook/commands/FindCommand.java [hunk 2]
```diff
@@ -48,13 +49,26 @@ public CommandResult execute() {
...
                 matchedPersons.add(person);
             }
         }
         return matchedPersons;
     }
 
+    private Set<String> getLowerCaseWords(Collection<String> stringSet) {
+        Set<String> lowerCaseWords = new HashSet<>();
+        for (String str: stringSet)
+        {
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/FindCommand.java:66]: Coding standard violation here.

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -83,6 +83,14 @@ public void addPerson(Person toAdd) throws DuplicatePersonException {
         syncTagsWithMasterList(toAdd);
     }
 
+    /**
+     * Sort every person in address book
+     */
+    public void sort()
+    {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/AddressBook.java:90]: Again, coding standard violation here.

File: src/seedu/addressbook/data/person/Contact.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Input validation purpose
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:4]: Irrelevant header comment

File: src/seedu/addressbook/data/person/Contact.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+public class Contact {
+
+    public final String value;
+    public boolean isPrivate;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:10]: Try to make it private. 
In general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Email.java [hunk 6]
```diff
@@ -6,28 +6,23 @@
  * Represents a Person's email in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidEmail(String)}
  */
-public class Email {
+public class Email extends Contact{
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Email.java:9]: Need space between `Contact` and `{`

File: test/input.txt [hunk 7]
```diff
@@ -101,6 +101,7 @@
 
   # find unique keyword
   find Betsy
+  find betsy
```
Comment (okkhoy) [inline test/input.txt:104]: Where are the tests for the sort command?

**Review by okkhoy — COMMENTED** [review]
1. Please pay attention to coding standards
2. You need to update the tests for sort command and include appropriate documentation describing the feature
3. Also, please check why your tests are failing.
Please update the PR before next tutorial.

**Comment by okkhoy** [comment]
@AJZ1995 
Some comments added. Please ack & update the PR after reading comments.

**Comment by AJZ1995** [comment]
Ack

**Pull request closed without merging**